Sources/SwiftDocCUtilities/Action/Actions/MergeAction.swift

```diff
@@ -0,0 +1,65 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+@_spi(FileManagerProtocol) import SwiftDocC
+
+/// An action that merges a list of documentation archives into a combined archive.
+struct MergeAction: Action {
+    var archives: [URL]
+    var landingPageCatalog: URL?
+    var outputURL: URL
+    var fileManager: FileManagerProtocol
+
+    mutating func perform(logHandle: LogHandle) throws -> ActionResult {
+        guard let firstArchive = archives.first else {
+            // A validation warning should have already been raised in `Docc/Merge/InputAndOutputOptions/validate()`.
+            return ActionResult(didEncounterError: true, outputs: [])
+        }
+        
+        try? fileManager.removeItem(at: outputURL)
+        try fileManager.copyItem(at: firstArchive, to: outputURL)
```

Comment:
Might be a good idea to create a temporary directory here, and use that to build up the result, and then copy/move it to the final location when and if the entire merge operation is successful. Otherwise, the user would end up with a partially merged but invalid result if there was some error or exception during the merge processing.

Reply:
Fixed in https://github.com/apple/swift-docc/pull/821/commits/f98be845a62d75c7d8d019587e09793abd0ad12a